src/Microsoft.Health.Fhir.Core.UnitTests/Features/Operations/Import/ImportProcessingTaskTests.cs

```diff
@@ -0,0 +1,398 @@
+                });
+
+            ImportProcessingTask task = new ImportProcessingTask(
+                                    inputData,
+                                    progress,
+                                    bulkOperation,
+                                    loader,
+                                    importer,
+                                    importErrorStoreFactory,
+                                    contextUpdater,
+                                    contextAccessor,
+                                    loggerFactory);
+
+            await Assert.ThrowsAsync<RetriableTaskException>(() => task.ExecuteAsync());
+
+            ImportProcessingProgress progressForContext = JsonConvert.DeserializeObject<ImportProcessingProgress>(context);
+            Assert.Equal(progressForContext.CurrentIndex, currentIndex);
+        }
+
+        [Fact]
+        public async Task GivenImportTaskInput_WhenExceptionThrowForLoad_ThenRetriableExceptionShouldBeThrow()
+        {
+            ImportProcessingTaskInputData inputData = GetInputData();
+            ImportProcessingProgress progress = new ImportProcessingProgress();
+
+            IFhirDataBulkImportOperation bulkOperation = Substitute.For<IFhirDataBulkImportOperation>();
+            IImportResourceLoader loader = Substitute.For<IImportResourceLoader>();
+            IResourceBulkImporter importer = Substitute.For<IResourceBulkImporter>();
+            IImportErrorStore importErrorStore = Substitute.For<IImportErrorStore>();
+            IImportErrorStoreFactory importErrorStoreFactory = Substitute.For<IImportErrorStoreFactory>();
+            IContextUpdater contextUpdater = Substitute.For<IContextUpdater>();
+            RequestContextAccessor<IFhirRequestContext> contextAccessor = Substitute.For<RequestContextAccessor<IFhirRequestContext>>();
+            ILoggerFactory loggerFactory = new NullLoggerFactory();
+
+            loader.LoadResources(Arg.Any<string>(), Arg.Any<long>(), Arg.Any<string>(), Arg.Any<Func<long, long>>(), Arg.Any<CancellationToken>())
+                .Returns(callInfo =>
+                {
+                    long startIndex = (long)callInfo[1];
+                    Func<long, long> idGenerator = (Func<long, long>)callInfo[3];
+                    Channel<ImportResource> resourceChannel = Channel.CreateUnbounded<ImportResource>();
+
+                    Task loadTask = Task.Run(() =>
+                    {
+                        try
+                        {
+                            throw new InvalidOperationException();
+                        }
+                        finally
+                        {
+                            resourceChannel.Writer.Complete();
+                        }
+                    });
+
+                    return (resourceChannel, loadTask);
+                });
+
+            importer.Import(Arg.Any<Channel<ImportResource>>(), Arg.Any<IImportErrorStore>(), Arg.Any<CancellationToken>())
```

Comment:
Nit: the import and updateContextAsync method mocks aren't needed in this test since the exception is thrown before they are called.